builtin/mainmenu/settings/dlg_settings.lua

```diff
@@ -414,7 +414,7 @@ local function get_formspec(dialogdata)
 
 		fs[#fs + 1] = "style_type[image_button;border=false;padding=]"
 
-		local show_reset = comp.changed and comp.setting and comp.setting.default
+		local show_reset = comp.resettable and comp.setting and comp.setting.default
```

Comment:
It looks to me like the settingtypes parser always adds a default. But in case it's not there, there should still be a reset button.
Maybe something like this:
```suggestion
		local show_reset = comp.resettable
		local default = comp.setting and comp.setting.default or ""
```

Reply:
kept the `comp.setting` condition because `comp.setting.name` is needed for resetting